src/screens/pipelineScreen/index.js

```diff
@@ -1,41 +1,302 @@
+    //     const errorCode = error.code;
+    //     const errorMessage = error.message;
+    //     alert(errorCode + " " + errorMessage);
+    //   });
+    setEmail("");
+  }
+
+  // *******************************  CREATE TASK **************************************
+  const [task, setTask] = useState("");
+  const [labels, setLabels] = useState([""]);
+
+  async function createTask() {
+    if (task.length === 0) {
+      alert("Please enter a task");
+      return;
+    }
+    if (task.indexOf("/") !== -1) {
+      alert("Task cannot contain '/'.");
+      return;
+    }
+    if (labels.includes("")) {
+      alert("Can't add empty label(s)!");
+      return;
+    }
+
+    const ref = doc(db, "tasks", task);
+    const docSnap = await getDoc(ref);
+
+    if (docSnap.exists()) {
+      // task exists
+      const doUpdate = window.confirm("Task already exists. Update labels?");
+      if (doUpdate) {
+        try {
+          await updateDoc(ref, {
+            labels: arrayUnion(...labels),
+          });
+        } catch (e) {
+          alert(e);
+        }
+      }
+    } else {
+      // task does not exist, create new task
+      try {
+        //add to task collection
+        await setDoc(ref, {
+          labels: labels,
+        });
+        //add to taskList
+        await updateDoc(doc(db, "lists", "taskList"), {
+          tasks: arrayUnion(task),
+        });
+      } catch (e) {
+        alert(e);
+      }
+    }
+
+    // clear fields
+    setLabels([""]);
+    setTask("");
+  }
+
+  const showLabelsList = (labels) => {
+    return labels.map((l, idx) => {
+      return (
+        <div key={idx}>
+          <div class="field half">
+            <label for="name">Label Name</label>
+            <input
+              style={{ width: "50%" }}
+              type="text"
+              name="label"
+              id="label"
+              value={labels[idx]}
+              onChange={(e) => {
+                var edited = [...labels];
+                edited[idx] = e.target.value;
+                setLabels(edited);
+              }}
+            />
+            {idx === 0 ? (
+              <></>
+            ) : (
+              <>
+                <br />
+                <button
+                  onClick={() => {
+                    var removed = [...labels];
+                    removed.splice(idx, 1);
+                    setLabels(removed);
+                  }}
+                >
+                  remove
+                </button>
+              </>
+            )}
+          </div>
+          <br />
+        </div>
+      );
+    });
+  };
+
+  // *******************************  ASSIGN TASK **************************************
+  const [assignByUserId, setAssign] = useState(true);
+  const [usersList, setUsersList] = useState([]);
+  const [groupsList, setGroupsList] = useState([]);
+  const [tasksList, setTasksList] = useState([]);
+  // const [assignToUser, setAssignToUser] = useState("");
```

Comment:
Why are these commented out? Will you need it later or can we remove?

Reply:
oops sorry I meant to delete them.